### PR TITLE
clientconfig: Support for clouds-public.yaml

### PR DESCRIPTION
**Pull request by haxorof**
Here is an updated version of what I did initially.

@jtopjian Please review and see if I shall change or add something.

As discussed this replaces PR #40

### CODE REVIEW

File: openstack/clientconfig/requests.go [hunk 1]
```diff
@@ -65,9 +65,26 @@ func LoadYAML() (map[string]Cloud, error) {
 	return clouds.Clouds, nil
 }
 
+// LoadCloudsPublicYAML will load a clouds.yaml file and return the full config.
+func LoadCloudsPublicYAML() (map[string]PublicCloud, error) {
```
[the remark this answers is not in the document]
Comment (haxorof) [reply openstack/clientconfig/requests.go]: Will change that! Thanks!

File: openstack/clientconfig/utils.go [hunk 2]
```diff
@@ -6,9 +6,30 @@ import (
...
+// for strings.
+func defaultIfEmpty(value string, defaultValue string) string {
+	if value == "" {
+		return defaultValue
+	}
+	return value
+}
+
+// updateAuthInfo updates AuthInfo in Cloud struct with AuthInfo in PublicCloud.
+func updateAuthInfo(cloud *Cloud, publicCloud *PublicCloud) {
```
[the remark this answers is not in the document]
Comment (haxorof) [reply openstack/clientconfig/utils.go]: Thanks! 😄

**Comment by haxorof** [comment]
Now it is updated according to your comments.
Cheers! 😄

**Comment by jtopjian** [comment]
LGTM - thank you :)

**Pull request merged**